### PR TITLE
To Do list: interactive list

### PR DESCRIPTION
**Pull request by anuarshaidenov**
## Milestone details:
- [x] Created a separate file `listView.js`.
- [x] Created a class `ToDoList` to control the list.
- [x] Created a private method `#storeData()` to store the list data in the local storage.
- [x] Created a private method `#updateTaskStatus` to update, store and display each task everytime the checkbox gets clicked.

### CODE REVIEW

File: src/js/listView.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  #updateTaskStatus(task) {
+    const { id } = task.dataset;
+    const itemToChange = this.#tasks.find((task) => task.id === id);
+    if (task.checked) {
+      itemToChange.completed = true;
+    } else {
+      itemToChange.completed = false;
+    }
+    this.#storeData();
+    this.#displayList();
```
Comment (dasileker) [inline src/js/listView.js:46]: - [ ] The update for the completed task is not correctly set, I do not get the data after checking the box.

Capture d’écran 2021-12-01 105409

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @anuarshaidenov,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

- after checking the box the complete task should be true but still getting false.

Capture d’écran 2021-12-01 105738


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/js/listView.js (hunk 1 above)
Comment (anuarshaidenov) [reply src/js/listView.js:46]: @dasileker I can see that you've set the local storage values yourself. As you can see below each item also has an `id` entry and it looks like you didn't set any value for that. Please, try to set the values properly and see if works
code

Comment (dasileker) [reply src/js/listView.js:46]: It should not be generated randomly, because7once you're supposed to remove the target book you can't remove it randomly, so you need to loop for each to give a ascendant number to each book added.

Comment (anuarshaidenov) [reply src/js/listView.js:46]: @dasileker the `id` is different from the `index`. To mark each task as completed I'm using a unique `id` and to sort the items in ascending order I'm using `index`. Are you suggesting I use `index` to know which item to remove?

Comment (dasileker) [reply src/js/listView.js:46]: Yes, you can since the index is working like id.

Comment (anuarshaidenov) [reply src/js/listView.js:46]: @dasileker okay, thanks

**Review by dasileker — APPROVED** [review]
## **Hi @anuarshaidenov,** 

### **Impressive Job 👏**

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 
Approved


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**